### PR TITLE
Platform singleton

### PR DESCRIPTION
**Pull request by ricab**
Create a `Platform` singleton to facilitate mocking. Move `get_network_interfaces_info()` into it. Provide a `MockPlatform` for it.

Note: I kept it in the `multipass::platform` namespace for now, for consistency with the free functions. When/if we move all free functions to the class, we should probably get rid of the `platform` namespace altogether, to reduce verbosity in things like `using multipass::platform::Platform::Platform` :)

### CODE REVIEW

**Review by townsend2010 — COMMENTED** [review]
This looks good to me.  I'm not going to block on it, but would you be able to just add a simple test in `tests/linux/test_platform_linux.cpp` that calls `get_network_interfaces_info()` and expects that exception type?  It will help with coverage and should be easy to add.  I'll see what you say before doing the final approval.

**Comment by ricab** [comment]
Sure, I can do that. It will be removed in #1960, but that way this PR is fully independent.

**Comment by townsend2010** [comment]
Oh, then it doesn't seem to make sense to add that test since that will just be removed in the child PR.

**Review by townsend2010 — APPROVED** [review]
I like it!  Thanks!

bors r+

**Comment by townsend2010** [comment]
Running bors again to update `staging`.

bors r+

**Comment by townsend2010** [comment]
The Mac and Windows build will fail, so merging manually.

**Pull request merged**
